webapp/utils/subgraph.ts

```diff
@@ -61,10 +92,12 @@ export const getLastIndexedBlock = function (chainId: Chain['id']) {
       'Content-Type': 'application/json',
     },
     method: 'POST',
-  }).then(
-    (r: GraphResponse<{ _meta: { block: { number: number } } }>) =>
-      r.data._meta.block.number,
-  )
+  })
+    .then(response => ensureJsonParsed(response))
+    .then(
+      (response: GraphResponse<{ _meta: { block: { number: number } } }>) =>
+        checkGraphQLErrors(response)._meta.block.number,
```

Comment:
this function is confusing from the outside. If the function checks for errors, then it should not return anything there are no errors. Perhaps it should be something like 

```suggestion
        checkGraphQLErrors(response)
        return response.data._meta.block.number,
```